python/ray/autoscaler/v2/scheduler.py

```diff
@@ -1596,16 +1596,26 @@ def _enforce_idle_termination(
                 # The node is not idle for too long, skip it.
                 continue
 
+            if node.sched_requests[ResourceRequestSource.PENDING_DEMAND]:
+                # The node is needed by the pending requests.
+                # Skip it.
```

Comment:
If a node is needed by pending requests, we don't termination it in this `_enforce_idle_termination` function.